### PR TITLE
PWN-7114 - New swap. Basic token list design

### PR DESCRIPTION
**Pull request by gslevinkov**
## Jira Ticket

https://p2pvalidator.atlassian.net/browse/PWN-7114

## Description of Work

- [x] add basic token and section header list

todo:  
- [x] add hidable search view and check that it works
- [x] add item decoration with white background and rounded corners
- [x] add other items without price 

## Screenshots (Optional)

| Screenshot #1 |
|:-------------:|
|Screen Shot 2023-02-19 at 18 47 51|

### CODE REVIEW

File: ui-kit/src/main/res/menu/menu_ui_kit_toolbar_search.xml [hunk 1]
```diff
@@ -9,4 +9,10 @@
         app:actionLayout="@layout/item_toolbar_search"
         app:showAsAction="always" />
 
+    <item
+        android:id="@+id/menuItemSearchIcon"
+        android:title="@string/common_search"
+        android:icon="@drawable/ic_search"
+        app:showAsAction="always" />
+
```
Comment (gslevinkov) [inline ui-kit/src/main/res/menu/menu_ui_kit_toolbar_search.xml:17]:
```suggestion
```

File: app/src/main/java/org/p2p/wallet/swap/ui/jupiter/tokens/presenter/SwapTokensMapper.kt [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+class SwapTokensMapper {
+    fun toCellItems(
+        currentSelectedTokenMint: Base58String,
+        userTokens: List<Token.Active>,
+        swapTokens: List<JupiterSwapToken>
+    ): List<AnyCellItem> = buildList {
+        addAll(
+            chosenTokenGroup(
+                tokenIconUrl = (
+                    "https://raw.githubusercontent.com/solana-labs/token-list/main/assets/mainnet/" +
```
Comment (kabramovp2p) [inline app/src/main/java/org/p2p/wallet/swap/ui/jupiter/tokens/presenter/SwapTokensMapper.kt:31]: why it is hardcoded?

Comment (gslevinkov) [reply app/src/main/java/org/p2p/wallet/swap/ui/jupiter/tokens/presenter/SwapTokensMapper.kt:31]: for testing purposes, no logic here yet

File: app/src/main/java/org/p2p/wallet/swap/ui/jupiter/tokens/presenter/SwapTokensPresenter.kt [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+) : BasePresenter<SwapTokensContract.View>(), SwapTokensContract.Presenter {
+
+    override fun attach(view: SwapTokensContract.View) {
+        super.attach(view)
+        view.setTokenItems(
+            mapper.toCellItems(Base58String(""), emptyList(), emptyList())
+        )
+    }
+
+    override fun onSearchTokenQueryChanged(newQuery: String) {
```
Comment (kabramovp2p) [inline app/src/main/java/org/p2p/wallet/swap/ui/jupiter/tokens/presenter/SwapTokensPresenter.kt:19]: = Unit

File: app/src/main/java/org/p2p/wallet/swap/ui/jupiter/tokens/presenter/SwapTokensPresenter.kt [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+    }
+
+    override fun onTokenClicked() {
```
Comment (kabramovp2p) [inline app/src/main/java/org/p2p/wallet/swap/ui/jupiter/tokens/presenter/SwapTokensPresenter.kt:22]: = Unit

File: app/src/main/res/layout/fragment_jupiter_swap_tokens.xml [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+        android:layout_height="0dp"
+        android:orientation="vertical"
+        tools:listitem="@layout/item_finance_block"
+        app:layoutManager="androidx.recyclerview.widget.LinearLayoutManager"
+        app:layout_constraintBottom_toBottomOf="parent"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toBottomOf="@+id/toolbar" />
+
+</androidx.constraintlayout.widget.ConstraintLayout>
```
Comment (kabramovp2p) [inline app/src/main/res/layout/fragment_jupiter_swap_tokens.xml:33]: new empty line at the end of file

Comment (gslevinkov) [inline app/src/main/res/layout/fragment_jupiter_swap_tokens.xml:33]:
```suggestion
</androidx.constraintlayout.widget.ConstraintLayout>

```

File: app/src/main/java/org/p2p/wallet/swap/ui/jupiter/tokens/presenter/SwapTokensPresenter.kt (hunk 3 above)
Comment (gslevinkov) [reply app/src/main/java/org/p2p/wallet/swap/ui/jupiter/tokens/presenter/SwapTokensPresenter.kt:19]: no need for that now, this methods will be changed soon

File: app/src/main/java/org/p2p/wallet/swap/ui/jupiter/tokens/presenter/SwapTokensMapper.kt [hunk 6]
```diff
@@ -0,0 +1,159 @@
...
+
+        val tokenNameText = TextViewCellModel.Raw(
+            TextContainer.Raw(tokenName),
+            textAppearance = R.style.UiKit_TextAppearance_SemiBold_Text3,
+            textColor = R.color.text_night
+        )
+        val symbolOrAmountText = totalTokenAmount?.let { "$totalTokenAmount $tokenName" } ?: tokenName
+        val tokenSymbolWithAmount = TextViewCellModel.Raw(
+            TextContainer.Raw(symbolOrAmountText),
+            textAppearance = R.style.UiKit_TextAppearance_Regular_Label1,
```
[the remark this answers is not in the document]
Comment (gslevinkov) [reply app/src/main/java/org/p2p/wallet/swap/ui/jupiter/tokens/presenter/SwapTokensMapper.kt:135]: fixed. thanks!

**Review by NikolaiEmelianov — APPROVED** [review]

**Review by kabramovp2p — APPROVED** [review]

**Pull request merged**